db/sg_replicate_cfg_test.go

```diff
@@ -213,3 +214,176 @@ func TestReplicateManagerConcurrentReplicationOperations(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, 0, len(replications))
 }
+
+func TestRebalanceReplications(t *testing.T) {
+
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyReplicate)()
+
+	type rebalanceTest struct {
+		name                  string                     // Test name
+		nodes                 map[string]*SGNode         // Initial node set
+		replications          map[string]*ReplicationCfg // Initial replication assignment
+		expectedMinPerNode    int                        // Minimum replications per node after rebalance
+		expectedMaxPerNode    int                        // Maximum replications per node after rebalance
+		expectedTotalAssigned int                        // Expected total number of assigned replications post-rebalance
+	}
+	testCases := []rebalanceTest{
+		{
+			name: "new nodes",
+			nodes: map[string]*SGNode{
+				"n1": {Host: "n1"},
+				"n2": {Host: "n2"},
+				"n3": {Host: "n3"},
+			},
+			replications: map[string]*ReplicationCfg{
+				"r1": {ID: "r1", AssignedNode: "n1"},
+				"r2": {ID: "r2", AssignedNode: "n1"},
+				"r3": {ID: "r3", AssignedNode: "n1"},
+			},
+			expectedMinPerNode:    1,
+			expectedMaxPerNode:    1,
+			expectedTotalAssigned: 3,
+		},
+		{
+			name: "new replications",
+			nodes: map[string]*SGNode{
+				"n1": {Host: "n1"},
+				"n2": {Host: "n2"},
+				"n3": {Host: "n3"},
+			},
+			replications: map[string]*ReplicationCfg{
+				"r1": {ID: "r1", AssignedNode: ""},
+				"r2": {ID: "r2", AssignedNode: ""},
+				"r3": {ID: "r3", AssignedNode: ""},
+			},
+			expectedMinPerNode:    1,
+			expectedMaxPerNode:    1,
+			expectedTotalAssigned: 3,
+		},
+		{
+			name: "remove nodes",
+			nodes: map[string]*SGNode{
+				"n1": {Host: "n1"},
+				"n2": {Host: "n2"},
+			},
+			replications: map[string]*ReplicationCfg{
+				"r1": {ID: "r1", AssignedNode: "n1"},
+				"r2": {ID: "r2", AssignedNode: "n2"},
+				"r3": {ID: "r3", AssignedNode: "n3"},
+				"r4": {ID: "r4", AssignedNode: "n4"},
+			},
+			expectedMinPerNode:    2,
+			expectedMaxPerNode:    2,
+			expectedTotalAssigned: 4,
+		},
+		{
+			name:  "no nodes",
+			nodes: map[string]*SGNode{},
+			replications: map[string]*ReplicationCfg{
+				"r1": {ID: "r1", AssignedNode: "n1"},
+				"r2": {ID: "r2", AssignedNode: "n1"},
+				"r3": {ID: "r3", AssignedNode: "n2"},
+			},
+			expectedMinPerNode:    0,
+			expectedMaxPerNode:    0,
+			expectedTotalAssigned: 0,
+		},
```

Comment:
Would be good to get a single-node test in here for coverage.

```suggestion
		},
		{
			name:  "single node",
			nodes: map[string]*SGNode{
				"n1": {Host: "n1"},
			},
			replications: map[string]*ReplicationCfg{
				"r1": {ID: "r1", AssignedNode: "n1"},
				"r2": {ID: "r2", AssignedNode: "n2"},
				"r3": {ID: "r3", AssignedNode: ""},
			},
			expectedMinPerNode:    3,
			expectedMaxPerNode:    3,
			expectedTotalAssigned: 3,
		},
```

Reply:
Makes sense - thanks.